test/core/simd/meta/simd_i64x2_arith.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+
+"""
+Generate i64x2 integer arithmetic operation cases.
+"""
+
+from simd_arithmetic import SimdArithmeticCase
+
+
+class SimdI64x2ArithmeticCase(SimdArithmeticCase):
+
+    LANE_LEN = 2
+    LANE_TYPE = 'i64x2'
+
+    @property
+    def hex_binary_op_test_data(self):
+        return [
+            ('0x3fffffffffffffff', '0x4000000000000000'),
+            ('0x4000000000000000', '0x4000000000000000'),
+            ('-0x3fffffffffffffff', '-0x40000000fffffff'),
+            ('-0x4000000000000000', '-0x400000000000000'),
+            ('-0x4000000000000000', '-0x400000000000001'),
+            ('0x7fffffffffffffff', '0x7ffffffffffffff'),
+            ('0x7fffffffffffffff', '0x01'),
+            ('0x8000000000000000', '-0x01'),
+            ('0x7fffffffffffffff', '0x8000000000000000'),
+            ('0x8000000000000000', '0x8000000000000000'),
+            ('0xffffffffffffffff', '0x01'),
+            ('0xffffffffffffffff', '0xffffffffffffffff')
+        ]
+
+    @property
+    def hex_unary_op_test_data(self):
+        return ['0x01', '-0x01', '-0x8000000000000000', '-0x7fffffffffffffff',
+                '0x7fffffffffffffff', '0x8000000000000000', '0xffffffffffffffff']
+
+    @property
+    def i64x2_i8x16_test_data(self):
+        return {
+            'i64x2.add': [
+                [['0x7fffffffffffffff', ['0', '0', '0', '0', '0', '0', '0', '0x80'] * 4], '-1',
+                 ['i64x2', 'i8x16', 'i64x2']],
+                [['1', '255'], '0', ['i64x2', 'i8x16', 'i64x2']]
```

Comment:
Is there documentation anywhere on how these structures are interpreted? As it is they seem very magical.

Reply:
It's interpreted in https://github.com/WebAssembly/simd/blob/master/test/core/simd/meta/simd.py#L26.

The first element `['1', '255']` presents the arguments of the assertion, the second element `'0'` presents the test expected result. the last element `['i64x2', 'i8x16', 'i64x2']` presents v128 interpretations of previous elements with one-one correspondence. If the value of each lane are the same, the script will splat the argument to a v128 with corresponding interpretations. 
i.e. this code will be interpreted as `[(v128.const i64x2 1,1), (v128 i8x16 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255), (v128.const i64x2 0 0)]`

This looks magical, but simplified the script a lot.

Reply:
I see, thanks. It would be good to add a comment in the code pointing to where they are interpreted to make it easier for readers to find out where they need to look.

Reply:
Agree! Done.